### DISCUSSION TITLE
Possible support for Arduino/"Remote play"

### DISCUSSION THREAD

**Discussion by escequi** [ndeadly/MissionControl#507] [Ideas]
Hello, i've been playing quite a few games with my friends through steam remote play, and i was wondering how doable would it be to make my Switch a "remote play" station by streaming the video/audio with my capture card, whats left would be getting an interface to input controls remotely - like using an arduino to output controller-like signals through bluetooth, getting it's input from a network client at my friend's pc (client on friend's PC would detect input from their controller, send it through network to my server, which will tell the arduino to send the input to MissionControl as if it's a controller), or maybe a possibility to do it with other system modules directly with usb (sys-con)

**Comment by ndeadly** [comment]
So you want to run a game on your switch, stream the video output to your friends, and have them send back inputs to your console remotely via your PC? (parsec for Switch would be nice)

The advantage of having something like an Arduino make a bluetooth connection would be that it could appear as an official controller to the console, and could be used on an unhacked Switch. Since you're asking here, I assume your Switch is both hacked, and you use Mission Control. In that case I think an Arduino is overkill, and an unneccesary link in the chain. 

I have thought about this kind of thing before, and about the idea of having Mission Control expose an interface for a LAN controller, so that you can establish a connection from a PC and send controller packets directly via TCP/UDP. Virtual controllers could then be attached via the debug pad API (sys-con uses this). Then I would provide a client application/library for interacting with it from the PC side. This probably wouldn't be too crazy to implement, but I have other higher priority things like BLE and USB support to work on at the moment.

**Reply by escequi** [reply]
> So you want to run a game on your switch, stream the video output to your friends, and have them send back inputs to your console remotely via your PC? (parsec for Switch would be nice)
> 
> The advantage of having something like an Arduino make a bluetooth connection would be that it could appear as an official controller to the console, and could be used on an unhacked Switch. Since you're asking here, I assume your Switch is both hacked, and you use Mission Control. In that case I think an Arduino is overkill, and an unneccesary link in the chain.
> 
> I have thought about this kind of thing before, and about the idea of having Mission Control expose an interface for a LAN controller, so that you can establish a connection from a PC and send controller packets directly via TCP/UDP. Virtual controllers could then be attached via the debug pad API (sys-con uses this). Then I would provide a client application/library for interacting with it from the PC side. This probably wouldn't be too crazy to implement, but I have other higher priority things like BLE and USB support to work on at the moment.

Yep basically this, i'm a rookie coder so i have no idea how to implement this (i could prob do the friend-me client/server but i have absolutely no idea how the switch work so i can't mess with the module itself), but i figured it wouldn't be too crazy to do, worth the suggestion i guess, hopefully in the future someone (or you :D) will work on it and good job so far on the module :)
